plugins/inputs/smart/smart.go

```diff
@@ -300,82 +357,41 @@ func gatherDisk(acc telegraf.Accumulator, usesudo, collectAttributes bool, smart
 			}
 		} else {
 			if collectAttributes {
-				if startStop := sasStartStopAttr.FindStringSubmatch(line); len(startStop) > 1 {
-					tags["id"] = "4"
-					tags["name"] = "Start_Stop_Count"
-					i, err := strconv.ParseInt(strings.Replace(startStop[1], ",", "", -1), 10, 64)
-					if err != nil {
-						continue
-					}
-					fields["raw_value"] = i
+				if matches := sasNvmeAttr.FindStringSubmatch(line); len(matches) > 2 {
+					if matches[1] == "Critical Warning" {
+						var flags uint8
+						if _, err := fmt.Sscanf(matches[2], "0x%x", &flags); err != nil {
+							continue
+						}
 
-					acc.AddFields("smart_attribute", fields, tags)
-					continue
-				}
+						for flag, name := range nvmeCriticalWarnings {
+							tags["name"] = name
 
-				if powerCycle := nvmePowerCycleAttr.FindStringSubmatch(line); len(powerCycle) > 1 {
-					tags["id"] = "12"
-					tags["name"] = "Power_Cycle_Count"
-					i, err := strconv.ParseInt(strings.Replace(powerCycle[1], ",", "", -1), 10, 64)
-					if err != nil {
-						continue
-					}
-					fields["raw_value"] = i
-
-					acc.AddFields("smart_attribute", fields, tags)
-					continue
-				}
+							acc.AddFields("smart_attribute", map[string]interface{}{
+								"raw_value": flags&flag > 0,
```

Comment:
We won't be add this as a bool since we already have it as an integer, and InfluxDB won't allow us to have mixed types.  Since we are attempting to model the NVMe log as `smart_attributes`, I would just do it like this and document or link to the values in the README (see net_response for an example):
```
smart_attribute,name=critical_warning raw_value=0i
```

Often with enum style data we do something like this with a tag + integer field, but I don't think we should do it here since it doesn't fit into the existing model very well:
```
smart_attribute,name=critical\ warning,warning=reliability\ degraded warning_code=0i
```

Reply:
@danielnelson ahh I see what you mean! Do you think having it as an integer is a little hard to interpret? Since it represents so many states / combinations of flags? i.e. 8 means just `read only`, but 9 means `read only` and `available spare threshold exceeded`. I suppose no one is screaming for this warning fidelity at the moment, so I am probably just splitting hairs 😂 . They can atleast do something like alert when any critical warning occurs with a simple >0 check.

Reply:
That's a good point, this isn't an enum it's a bit set, so it wouldn't really work to pair it with a human readable name=foo tag.  I also agree, it's pretty difficult to deal with a large list of integers.  Still because this is a smart_attribute, I think we should start with just the single field, but read on for more discussion of booleans and bit sets.

One reason we have often favored integer values over booleans is that Grafana/Chronograf aren't very good at displaying booleans outside of tables.  UI support is always changing though, so it may be worth looking into what support there is now.  Many of the outputs also do not support booleans, and require users to use the `enum` processor.

I think there are 2 things that a user would be interested in with a bitset: is the item in an error state (non-zero), and what flags are set.  Checking for an error state is not always something that makes sense depending on the semantics of the bit set, but I think it often does.  If we were going to encode a bit set for InfluxDB, we could do it as multiple fields, one for the bit set encoded an integer, and one for each flag in the bit set:
```
foo critical_warning=3i,critical_warning_spare_threshold=true,critical_warning_temperature_above_or_under_threshold=true
```

However, this might be tricky to query, since you would need to grab all bit set fields and check each one to see if it is set.  Probably only something you want to do if you are checking for a particular flag.  It occurs to me that the set type idea for tags we discussed could help here, imagine:
```
foo,critical_warning=spare_threshold,critical_warning=temperature_above_or_under_threshold critical_warning=3i
```

Reply:
👍 I like it. I will simplify critical warning to just an int for now then.